dm/dm/master/openapi.go

```diff
@@ -521,10 +521,36 @@ func (s *Server) DMAPIDeleteTask(c *gin.Context, taskName string, params openapi
 }
 
 // DMAPIGetTaskList url is:(GET /api/v1/tasks).
-func (s *Server) DMAPIGetTaskList(c *gin.Context) {
+func (s *Server) DMAPIGetTaskList(c *gin.Context, params openapi.DMAPIGetTaskListParams) {
 	// get sub task config by task name task name->source name->subtask config
 	subTaskConfigMap := s.scheduler.GetSubTaskCfgs()
 	taskList := config.SubTaskConfigsToOpenAPITask(subTaskConfigMap)
+
+	// fill status
+	if params.WithStatus != nil && *params.WithStatus {
+		// get source list for all task
+		var sourceList []string
+		sourceNameM := make(map[string]struct{}) // use map to avoid duplicate source name
+		for _, task := range taskList {
+			for _, sourceConf := range task.SourceConfig.SourceConf {
+				sourceNameM[sourceConf.SourceName] = struct{}{}
+			}
+		}
+		for sourceName := range sourceNameM {
+			sourceList = append(sourceList, sourceName)
+		}
+		// get status from workers
+		workerStatusList := s.getStatusFromWorkers(c.Request.Context(), sourceList, "", false)
+		// fill status for every task
+		for idx := range taskList {
+			subTaskStatusList, err := getOpenAPISubtaskStatusByTaskName(taskList[idx].Name, workerStatusList)
+			if err != nil {
+				_ = c.Error(err)
+				return
+			}
+			taskList[idx].StatusList = &subTaskStatusList
```

Comment:
please add a integration test for this feature when we have started multiple tasks.

I started `test-shard`, `test`, `testAA` and now the number of `status_list` is not expected

```

            "status_list": [
                {
                    "load_status": {
                        "finished_bytes": 0,
                        "meta_binlog": "(mysql-bin.000001, 2568)",
                        "meta_binlog_gtid": "2c43df5b-5eea-11ec-aff5-0242ac110002:1-12",
                        "progress": "0.00 %",
                        "total_bytes": 315
                    },
                    "name": "testAA",
                    "source_name": "mysql-01",
                    "stage": "Paused",
                    "unit": "Load",
                    "unresolved_ddl_lock_id": "",
                    "worker_name": "worker2"
                },
                {
                    "load_status": {
                        "finished_bytes": 0,
                        "meta_binlog": "(mysql-bin.000001, 2520)",
                        "meta_binlog_gtid": "def3d483-5ede-11ec-b145-0242ac110003:1-11",
                        "progress": "0.00 %",
                        "total_bytes": 158
                    },
                    "name": "testAA",
                    "source_name": "mysql-02",
                    "stage": "Paused",
                    "unit": "Load",
                    "unresolved_ddl_lock_id": "",
                    "worker_name": "worker1"
                },
                {
                    "load_status": {
                        "finished_bytes": 0,
                        "meta_binlog": "(mysql-bin.000001, 2520)",
                        "meta_binlog_gtid": "def3d483-5ede-11ec-b145-0242ac110003:1-11",
                        "progress": "0.00 %",
                        "total_bytes": 158
                    },
                    "name": "testAA",
                    "source_name": "mysql-02",
                    "stage": "Paused",
                    "unit": "Load",
                    "unresolved_ddl_lock_id": "",
                    "worker_name": "worker1"
                },
                {
                    "load_status": {
                        "finished_bytes": 0,
                        "meta_binlog": "(mysql-bin.000001, 2520)",
                        "meta_binlog_gtid": "def3d483-5ede-11ec-b145-0242ac110003:1-11",
                        "progress": "0.00 %",
                        "total_bytes": 158
                    },
                    "name": "testAA",
                    "source_name": "mysql-02",
                    "stage": "Paused",
                    "unit": "Load",
                    "unresolved_ddl_lock_id": "",
                    "worker_name": "worker1"
                },
                {
                    "load_status": {
                        "finished_bytes": 0,
                        "meta_binlog": "(mysql-bin.000001, 2568)",
                        "meta_binlog_gtid": "2c43df5b-5eea-11ec-aff5-0242ac110002:1-12",
                        "progress": "0.00 %",
                        "total_bytes": 315
                    },
                    "name": "testAA",
                    "source_name": "mysql-01",
                    "stage": "Paused",
                    "unit": "Load",
                    "unresolved_ddl_lock_id": "",
                    "worker_name": "worker2"
                },
                {
                    "load_status": {
                        "finished_bytes": 0,
                        "meta_binlog": "(mysql-bin.000001, 2568)",
                        "meta_binlog_gtid": "2c43df5b-5eea-11ec-aff5-0242ac110002:1-12",
                        "progress": "0.00 %",
                        "total_bytes": 315
                    },
                    "name": "testAA",
                    "source_name": "mysql-01",
                    "stage": "Paused",
                    "unit": "Load",
                    "unresolved_ddl_lock_id": "",
                    "worker_name": "worker2"
                }
            ],
```

Reply:
sorry could you please provied the task config?  i can't reproduce this case

the it test is added in https://github.com/pingcap/tiflow/pull/3971/commits/b05c993881abf0cd256c05f3b77575814c271ce6

Reply:
can't reproduce as well :joy: